src/kirby/components/modal/services/modal.helper.ts

```diff
@@ -74,8 +86,56 @@ export class ModalHelper {
     return Promise.resolve(
       baseAnimation
         .addElement(baseEl)
-        .easing('easeOut')
-        .duration(200)
+        .duration(duration)
+        .add(wrapperAnimation)
+        .add(backdropAnimation)
+    );
+  }
+
+  private static animateOut(
+    flavor: any,
+    duration: KirbyAnimation.Duration,
+    easing: KirbyAnimation.Easing,
+    AnimationC: Animation,
+    baseEl: HTMLElement
+  ): Promise<Animation> {
+    // Set-up animated elements
+    const baseAnimation = new AnimationC();
+    const backdropAnimation = new AnimationC();
+    backdropAnimation.addElement(baseEl.querySelector('ion-backdrop'));
+    const wrapperAnimation = new AnimationC();
+    const wrapperElem = baseEl.querySelector('.modal-wrapper') as HTMLElement;
+    wrapperAnimation.addElement(wrapperElem);
+
+    backdropAnimation.easing(KirbyAnimation.Easing.STATIC);
+    wrapperAnimation.easing(easing);
+
+    // Define animation transition values
+    let transformYFromTo = [`0px`, `${baseEl.clientHeight}px`];
+    let fadeBackdropFromTo = [0.3, 0.01];
+    let fadeWrapperFromTo = [1, 0.01];
+
+    // Define animations
+    if (flavor === 'drawer') {
+      // slide drawers up/down
+      backdropAnimation.fromTo('opacity', fadeBackdropFromTo[0], fadeBackdropFromTo[1]);
+      wrapperAnimation.beforeStyles({ opacity: 1 });
+      wrapperAnimation.fromTo(
+        `transform`,
+        `translateY(${transformYFromTo[0]})`,
+        `translateY(${transformYFromTo[1]})`
+      );
+    } else {
+      // fade modals in/out
+      backdropAnimation.fromTo('opacity', fadeBackdropFromTo[0], fadeBackdropFromTo[1]);
```

Comment:
Consider not using arrays for this as this makes the lines using these loosing their readability, unless you know the structure of the array.